### PR TITLE
TCD/JDJ/RULE19-13

### PR DESCRIPTION
**Pull request by JacksonJ-KC**
Getting schema error: "Schema validation in rule-19-13-(a,b,c,d) for the Baseline RMR: schema invalid: Additional properties are not allowed ('supply_design_cooling_setpoint_temperature', 'supply_design_heating_setpoint_temperature' were unexpected)"

So far have not been able to figure out what I am doing incorrectly.

### CODE REVIEW

**Comment by yunjoonjung-PNNL** [comment]
> Getting schema error: "Schema validation in rule-19-13-(a,b,c,d) for the Baseline RMR: schema invalid: Additional properties are not allowed ('supply_design_cooling_setpoint_temperature', 'supply_design_heating_setpoint_temperature' were unexpected)"
> 
> So far have not been able to figure out what I am doing incorrectly.

This is because ```supply_design_cooling_setpoint_temperature``` and ```supply_design_heating_setpoint_temperature``` don't exist in the current schema version. You can update the two keys from schema from [HERE](https://github.com/open229/ruleset-model-description-schema/blob/main/docs229/ASHRAE229.schema.json) (for some reason, copy to highlight doesn't work) into the current schema in your branch.

**Comment by yunjoonjung-PNNL** [comment]
Could we proceed this rule by updating the schema manually?

File: rct229/ruletest_engine/ruletest_jsons/ashrae9012019/section19/rule_19_13.json [hunk 1]
```diff
@@ -203,14 +216,16 @@
                                                     "fan_control": "VARIABLE_SPEED_DRIVE",
                                                     "supply_fans": [
                                                         {
-                                                            "id": "Supply Fan 1"
+                                                            "id": "Supply Fan 1",
+                                                            "design_airflow": 2359.737215999999
```
Comment (yunjoonjung-PNNL) [inline rct229/ruletest_engine/ruletest_jsons/ashrae9012019/section19/rule_19_13.json:220]: Do we need the ```is_airflow_autosized``` key here?

File: rct229/ruletest_engine/ruletest_jsons/ashrae9012019/section19/rule_19_13.json [hunk 2]
```diff
@@ -165,7 +166,19 @@
                                                         "type": "VARIABLE_AIR_VOLUME",
                                                         "served_by_heating_ventilating_air_conditioning_system": "System 7",
                                                         "heating_source": "HOT_WATER",
-                                                        "heating_from_loop": "Boiler Loop 1"
+                                                        "heating_from_loop": "Boiler Loop 1",
+                                                        "primary_airflow": 2359.737215999999,
+                                                        "supply_design_cooling_setpoint_temperature": 55,
```
Comment (yunjoonjung-PNNL) [inline rct229/ruletest_engine/ruletest_jsons/ashrae9012019/section19/rule_19_13.json:171]: Do we also need the ```supply_design_cooling_setpoint_temperature```, ```supply_design_heating_setpoint_temperature``` and ```spaces``` keys  for thermal zone 2?

Comment (JacksonJ-KC) [reply rct229/ruletest_engine/ruletest_jsons/ashrae9012019/section19/rule_19_13.json:171]: yes, thanks for catching

File: rct229/ruletest_engine/ruletest_jsons/ashrae9012019/section19/rule_19_13.json (hunk 1 above)
Comment (JacksonJ-KC) [reply rct229/ruletest_engine/ruletest_jsons/ashrae9012019/section19/rule_19_13.json]: Added!

File: rct229/ruletest_engine/ruletest_jsons/ashrae9012019/section19/rule_19_13.json [hunk 3]
```diff
@@ -181,7 +193,13 @@
                                                         "type": "VARIABLE_AIR_VOLUME",
                                                         "served_by_heating_ventilating_air_conditioning_system": "System 7",
                                                         "heating_source": "HOT_WATER",
-                                                        "heating_from_loop": "Boiler Loop 1"
+                                                        "heating_from_loop": "Boiler Loop 1",
+                                                        "primary_airflow": 2500
+                                                    }
+                                                ],
+                                                "spaces": [
+                                                    {
```
Comment (yunjoonjung-PNNL) [inline rct229/ruletest_engine/ruletest_jsons/ashrae9012019/section19/rule_19_13.json:201]: Do we need ```lighting_space_type``` for all the spaces?

Comment (JacksonJ-KC) [reply rct229/ruletest_engine/ruletest_jsons/ashrae9012019/section19/rule_19_13.json:201]: added

**Comment by jugonzal07** [comment]
@yunjoonjung-PNNL 's code implies there are two required fields for this rule and the [RDS ](https://github.com/pnnl/ruleset-checking-tool/blob/develop/docs/section19/Rule19-13.md)seems to imply it as well.

Here they are broken out by ruletest:
19-13
	• a
		· 'Terminal:VAV Air Terminal 2 is missing supply_design_cooling_setpoint_temperature'
	• b
		· 'id:VAV Fan System 1 missing:minimum_outdoor_airflow'
	• c
		· 'Terminal:VAV Air Terminal 2 is missing supply_design_cooling_setpoint_temperature'
	• d
		·'id:VAV Fan System 1 missing:minimum_outdoor_airflow'

**Comment by JacksonJ-KC** [comment]
> @yunjoonjung-PNNL 's code implies there are two required fields for this rule and the [RDS ](https://github.com/pnnl/ruleset-checking-tool/blob/develop/docs/section19/Rule19-13.md)seems to imply it as well.
> 
> Here they are broken out by ruletest: 19-13 • a · 'Terminal:VAV Air Terminal 2 is missing supply_design_cooling_setpoint_temperature' • b · 'id:VAV Fan System 1 missing:minimum_outdoor_airflow' • c · 'Terminal:VAV Air Terminal 2 is missing supply_design_cooling_setpoint_temperature' • d ·'id:VAV Fan System 1 missing:minimum_outdoor_airflow'

Thanks Juan. I found the issue for the missing `supply_design_cooling_setpoint_temperature` keys, however I do see the `minimum_outdoor_airflow` for VAV Fan System 1 in both ruletests b and d. Am I mistaken?

**Comment by JacksonJ-KC** [comment]
> > and 985. Am I mistaken?
> 
> Good point!
> 
> 1. Could you add the `minimum_outdoor_airflow` in the `proposed` part as well?
> 2. Could you add the `specification_method` and `design_electric_power` keys and appropriate their values?

@yunjoonjung-PNNL I believe you meant `primary_airflow` for the proposed so I made sure to include that for all cases. Let me know if you also need `minimum_outdoor_airflow` but I don't think that should be applicable to this rule. 

Also added `specification_method` and `design_electric_power` to the baseline fans for `get_fan_system_object_supply_return_exhaust_relief_total_power_flow()`

**Pull request closed without merging**